Automattic/wp-calypso
# Subscriptions Management: Add optimistic updates to all settings / toggles on the Individual Site Subscription page

Resolves https://github.com/Automattic/wp-calypso/issues/77174.

## Proposed Changes

Add optimistic update to:

* `NotifyMeOfNewPosts` toggle
* `EmailMeNewPosts` toggle
* `Delivery Frequency` selector
* `EmailMeNewComments` toggle

![Markup on 2023-05-22 at 16:39:28](https://github.com/Automattic/wp-calypso/assets/25105483/a828fbe6-3e6a-458a-a1ea-18ef23f72646)

**Please note**

ℹ️ This PR does not address optimistic updates for subscribe / unsubscribe actions. These require a bit different approach - since once you unsubscribe, information like Site title, Site icon and subscriptions count become unavailable - while we still need it for display. This will be addressed this in a separate task (https://github.com/Automattic/wp-calypso/issues/77093).

ℹ️ The `EmailMeNewComments` toggle always returns to the original state because there's a bug in the API which needs to be addressed separately (https://github.com/Automattic/wp-calypso/issues/77173).

## Testing Instructions

### Testing as an external user
1. Check out the PR and build the app.
2. Authenticate as an external user (`subkey` "dance" in the incognito browser window).
3. Make sure your external user is subscribed to some sites already.
4. Navigate to `http://calypso.localhost:3000/subscriptions/site/[blog_id]` where `[blog_id]` needs to be replaced with a specific blog ID your user is subscribed to.
5. When the page loads, you should be able to change the "Email delivery frequency" setting (this can be checked at https://wordpress.com/subscriptions/sites).

### Testing as an logged-in user
1. Check out the PR.
2. Navigate to `client/server/pages/index.js` and temporarily add `return next();` to the line `833`:

![Markup on 2023-04-28 at 17:50:44](https://user-images.githubusercontent.com/25105483/235195033-4902bb9d-db57-43a4-b295-2d90feb6731f.png)

3. Build the app.
4. Log in to a test WordPress.com account that has some existing Site subscriptions.
5. Navigate to `http://calypso.localhost:3000/subscriptions/site/[blog_id]` where `[blog_id]` needs to be replaced with a specific blog ID your user is subscribed to.
7. When the page loads, you should be able to switch all the toggles (this can be checked at https://wordpress.com/subscriptions/sites).

## Pre-merge Checklist
